backend/logic/deploy/cdn/aws.js

```diff
@@ -0,0 +1,601 @@
+} = require('../../../utils/const')
+const route53 = require('../../../utils/dns/route53')
+const { assert } = require('../../../utils/validators')
+const { getLogger } = require('../../../utils/logger')
+
+const log = getLogger('logic.deploy.cdn.aws')
+
+let cloudfrontClient, acmClient, awsCredentials, s3Client
+
+/**
+ * Check if AWS is configured and we can deploy to it
+ *
+ * @param args {Object}
+ * @param args.networkConfig {Object} - Decrypted networkConfig object
+ * @returns {bool} - if we can deploy
+ */
+function isAvailable({ networkConfig, resourceSelection }) {
+  return (
+    resourceSelection.includes('aws-cdn') &&
+    isConfigured(networkConfig, 'aws-cdn')
+  )
+}
+
+/**
+ * Configure this singleton for a use
+ *
+ * @param args {Object}
+ * @param args.networkConfig {Object} - Decrypted networkConfig object
+ */
+async function configure({ networkConfig }) {
+  awsCredentials = {
+    accessKeyId: networkConfig.awsAccessKeyId,
+    secretAccessKey: networkConfig.awsSecretAccessKey,
+    region: networkConfig.awsRegion || DEFAULT_AWS_REGION
+  }
+  cloudfrontClient = new CloudFront({
+    apiVersion: '2020-05-31',
+    region: networkConfig.awsRegion || DEFAULT_AWS_REGION,
+    ...awsCredentials
+  })
+  acmClient = new ACM({
+    apiVersion: '2015-12-08',
+    region: networkConfig.awsRegion || DEFAULT_AWS_REGION,
+    ...awsCredentials
+  })
+  s3Client = new S3({
+    apiVersion: '2006-03-01',
+    region: networkConfig.awsRegion || DEFAULT_AWS_REGION,
+    ...awsCredentials
+  })
+}
+
+/**
+ * Configure CDN to point to bucket
+ *
+ * @param args {Object}
+ * @param args.shop {Object} - Shop model instance
+ * @param args.deployment {Object} - ShopDeployment model instance
+ * @param args.domains {Array<string>} - name of Google Cloud Storage bucket
+ */
+async function configureCDN({ shop, deployment, domains }) {
+  assert(!!shop, 'shop must be provied')
+  assert(!!deployment, 'deployment must be provied')
+  assert(domains instanceof Array, 'domains must be an array')
+
+  if (!deployment.bucketUrls) {
+    log.warn(`No bucket URLs found in deployment`)
+    return
+  }
+
+  const urls = deployment.bucketUrls.split(',').map((u) => new URL(u))
+  const bucketUrl = _find(urls, (o) => o.protocol === 's3:')
+
+  if (!bucketUrl) {
+    log.warn(`No S3 bucket URL found in deployment`)
+    return
+  }
+
+  const bucketName = bucketUrl.hostname
+  const requestId = String(+new Date())
+
+  let acmCertificateArn
+  let cert = await getCertificate(domains[0])
+  if (cert) {
+    log.debug('Found Certificate')
+
+    acmCertificateArn = cert.CertificateArn
+  } else {
+    log.debug('Creating Certificate')
+
+    acmCertificateArn = await createCertificate({ requestId, domains })
+    cert = await getCertificateByARN(acmCertificateArn)
+  }
+
+  // Do DNS validation if necessary
+  for (const dvo of cert.DomainValidationOptions) {
+    const status = dvo.ValidationStatus
+
+    if (status === 'FAILED') {
+      console.log(dvo)
```

Comment:
use log.debug

Reply:
Explicitly wanted to dump the raw object without all the nice logging for this one.  But it makes me itch a little too.  Will change.